### PR TITLE
save message status in the DB, and more refactoring

### PR DESCRIPTION
**Pull request by tobiasmcnulty**
- Creates a new `MTMessageStatus` model to collect a variety of information about MT messages sent
- Refactors things a bit, trying not to let the client class get to big...

### CODE REVIEW

File: src/smpp_gateway/models.py [hunk 1]
```diff
@@ -62,3 +69,43 @@ def save(self, *args, **kwargs):
...
+        constraints = [
+            models.UniqueConstraint(
+                fields=["backend", "sequence_number"], name="unique_seq_num"
+            )
+        ]
+        index = [
+            models.UniqueConstraint(
+                fields=["backend", "sequence_number"], name="unique_seq_num"
+            )
+        ]
```
Comment (copelco) [inline src/smpp_gateway/models.py:111]:
```suggestion
```

runserver wouldn't run with this -- I think it's likey just a typo?

File: src/smpp_gateway/client.py [hunk 2]
```diff
@@ -57,47 +61,128 @@ class PgSmppClient(smpplib.client.Client):
...
+        params = decoded_params(pdu)
+        count = MTMessageStatus.objects.filter(
+            backend=self.backend,
+            sequence_number=pdu.sequence,
+        ).update(
+            modify_time=timezone.now(),
+            command_status=pdu.status,
+            message_id=params["message_id"] or "",
+        )
+        if count == 0:
```
Comment (copelco) [inline src/smpp_gateway/client.py:120]: 🆒

File: src/smpp_gateway/models.py [hunk 3]
```diff
@@ -35,8 +41,11 @@ def decoded_short_message(self):
     def __str__(self):
         return f"{self.params['short_message']} ({self.id})"
 
+    class Meta:
+        verbose_name = "mobile-originated message"
```
Comment (copelco) [inline src/smpp_gateway/models.py:45]: 💯

**Review by copelco — COMMENTED** [review]
lgtm - 1 possible bug in a `Meta` class, otherwise I say we merge!

File: src/smpp_gateway/models.py (hunk 1 above)
Comment (tobiasmcnulty) [reply src/smpp_gateway/models.py]: Good catch! Yeah, I was going to add a composite index..but you can see I didn't get very far with it. 😩

**Pull request merged**